models/Sensor.js

```diff
@@ -0,0 +1,150 @@
+const mongoose = require("mongoose");
+
+const sensorSchema = new mongoose.Schema(
+  {
+    // Now putting settings inside specs
+    specs: {
+      // Serial number is now auto-generated on BE
+      serialNumber: {
+        type: String,
+        trim: true,
+        maxlength: 255
+      },
+      electrodeType: {
+        type: String,
+        required: true,
+        maxlength: 255
+      },
+      printer: {
+        type: String,
+        required: true,
+        maxlength: 255
+      },
+      inkType: {
+        type: String,
+        required: true,
+        maxlength: 255
+      },
+      concentration: {
+        type: String,
+        required: true,
+        maxlength: 4
+      },
+      electrodeBatchDate: {
+        type: String,
+        required: true,
+        maxlength: 255
+      },
+      // now adding name (this will need to be calculated on BE)
+      // format for name is '${electrodeType}-${electrodePatchDate}-${Printer}-${inkType}-${concentraion}-${serialNum}'
+      name: { type: String }
+    },
+    // now we separate printing layer
+    printingLayers: {
+      a0: { type: String },
+      a1: { type: String },
+      a2: { type: String },
+      a3: { type: String },
+      a4: { type: String },
+      a5: { type: String },
+      a6: { type: String },
+      a7: { type: String }
+    },
+    // now we have seperate array for all of resistance history
+    resistanceSamplesHistory: [
+      {
+        date: { type: Date },
```

Comment:
instead of `date`, let's be more specific and say `createdAt`

Reply:
doneeeeeeeeeeeeeeeeeeeeee